crates/hir_def/src/nameres/tests/mod_resolution.rs

```diff
@@ -323,13 +323,26 @@ pub struct Baz;
 fn module_resolution_relative_path_outside_root() {
     check(
         r#"
-//- /main.rs
+//- /a/b/c/d/e/main.rs root:
 #[path="../../../../../outside.rs"]
 mod foo;
+
+//- /outside.rs
+mod bar;
+
+//- /bar.rs
```

Comment:
Hm, shouldn't it be `/outside/bar.rs`? I see that the test passes, so there might be a bug in rust-analyzer?

Reply:
Good catch, you're right that should be `/outside/bar.rs` or `/outside/bar/mod.rs`.  I'll see if I can figure out what's happening there.

Reply:
Err actually I think this behavior is correct, my bad.  If you declare a non-inline mod, it's located in the same parent as where it's declared.  `/main.rs: mod foo;` -> `/foo.rs`